### PR TITLE
Fiona copyrighter tool - development

### PR DESCRIPTION
**Pull request by Akyiaa**

### CODE REVIEW

**Comment by galasa-team** [comment]
Build successful

File: offline-tools/copyrighter/pkg/cmd/galasacopyrighter.go [hunk 1]
```diff
@@ -28,6 +28,8 @@ const (
 	COMMENT_START_JAVA    = "/*"
 	COMMENT_CONTINUE_JAVA = " *"
 	COMMENT_END_JAVA      = " */"
+	COMMENT_DOUBLE_STROKE = "//"
```
Comment (techcobweb) [inline offline-tools/copyrighter/pkg/cmd/galasacopyrighter.go:31]: This isn't being used, so needs to be removed.

File: offline-tools/copyrighter/pkg/cmd/galasacopyrighter.go [hunk 2]
```diff
@@ -28,6 +28,8 @@ const (
...
+	COMMENT_BASH_SCRIPT   = "#"
```
Comment (techcobweb) [inline offline-tools/copyrighter/pkg/cmd/galasacopyrighter.go:32]: Same here

File: offline-tools/copyrighter/pkg/cmd/galasacopyrighter_test.go [hunk 3]
```diff
@@ -190,3 +190,89 @@ func TestClosingCommentIsBeforeOpeningCommentFailsWithError(t *testing.T) {
...
+ * Copyright contributors to the Galasa project
+ */
+ package mypackage`
+
+	// When...
+	output, _ := setCopyright(input)
+
+	// Then..
+	assert.NotNil(t, output)
+	assert.Contains(t, output, "* SPDX-License-Identifier: EPL-2.0")
```
Comment (techcobweb) [inline offline-tools/copyrighter/pkg/cmd/galasacopyrighter_test.go:260]: Lets check that the output doesn't have `Copyright contributors to the Galasa project` inside ?

**Comment by galasa-team** [comment]
Build successful

File: offline-tools/copyrighter/pkg/cmd/galasacopyrighter.go (hunk 1 above)
Comment (Akyiaa) [reply offline-tools/copyrighter/pkg/cmd/galasacopyrighter.go:31]: removed

File: offline-tools/copyrighter/pkg/cmd/galasacopyrighter.go (hunk 2 above)
Comment (Akyiaa) [reply offline-tools/copyrighter/pkg/cmd/galasacopyrighter.go:32]: removed

File: offline-tools/copyrighter/pkg/cmd/galasacopyrighter.go [hunk 4]
```diff
@@ -87,7 +89,13 @@ func stripOutExistingCopyright(input string) (string, error) {
 
 				afterFirstCommentEnder := input[startIndexOfFollowingInput:]
 
-				output = beforeFirstCommentInput + afterFirstCommentEnder
+				//checking if the comment contains the old copyright text
+				commentToCheck := input[firstCommentOpener:startIndexOfFollowingInput]
+				if strings.Contains(commentToCheck, "Copyright ") {
```
[the remark this answers is not in the document]
Comment (Akyiaa) [reply offline-tools/copyrighter/pkg/cmd/galasacopyrighter.go:94]: space gone!

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build failed, see http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/galasa-build/pipelineruns/repo-automation-pr-269-chnnf for details. If you are unable to do so, please contact a member of the Galasa team.

**Comment by galasa-team** [comment]
Build failed, see http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/galasa-build/pipelineruns/repo-automation-pr-269-c6lhj for details. If you are unable to do so, please contact a member of the Galasa team.

**Comment by galasa-team** [comment]
Build failed, see http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/galasa-build/pipelineruns/repo-automation-pr-269-tprv6 for details. If you are unable to do so, please contact a member of the Galasa team.

**Comment by galasa-team** [comment]
Build failed, see http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/galasa-build/pipelineruns/repo-automation-pr-269-4ftqt for details. If you are unable to do so, please contact a member of the Galasa team.

**Comment by galasa-team** [comment]
Build failed, see http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/galasa-build/pipelineruns/repo-automation-pr-269-p2pnf for details. If you are unable to do so, please contact a member of the Galasa team.

**Comment by galasa-team** [comment]
Build failed, see http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/galasa-build/pipelineruns/repo-automation-pr-269-prxxx for details. If you are unable to do so, please contact a member of the Galasa team.

**Comment by galasa-team** [comment]
Build failed, see http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/galasa-build/pipelineruns/repo-automation-pr-269-prxxx-r-5z8s8 for details. If you are unable to do so, please contact a member of the Galasa team.

File: offline-tools/copyrighter/.gitignore [hunk 5]
```diff
@@ -1,4 +1,10 @@
+<<<<<<< HEAD
+build/
+bin/
+temp/
+=======
```
Comment (techcobweb) [inline offline-tools/copyrighter/.gitignore:5]: This is a merge conflict which needs resolving.

File: build-images/github-webhook-receiver/go.mod [hunk 6]
```diff
@@ -4,15 +4,17 @@ go 1.15
 
 require (
 	github.com/spf13/cobra v1.6.1
-	github.com/stretchr/testify v1.8.1
+	github.com/stretchr/testify v1.8.4
```
Comment (techcobweb) [inline build-images/github-webhook-receiver/go.mod:7]: Suggest you revert the file contents to what it used to be.

File: build-images/github-webhook-receiver/go.sum [hunk 7]
```diff
@@ -28,39 +28,135 @@ cloud.google.com/go v0.93.3/go.mod h1:8utlLll2EF5XMAV15woO4lSbWQlk8rer9aLOfLh7+Y
 cloud.google.com/go v0.94.1/go.mod h1:qAlAugsXlC+JWO+Bke5vCtc9ONxjQT3drlTTnAplMW4=
 cloud.google.com/go v0.97.0/go.mod h1:GF7l59pYBVlXQIBLx3a761cZ41F9bBH3JUlihCt2Udc=
 cloud.google.com/go v0.99.0/go.mod h1:w0Xx2nLzqWJPuozYQX+hFfCSI8WioryfRDzkoI/Y2ZA=
+cloud.google.com/go v0.100.1/go.mod h1:fs4QogzfH5n2pBXBP9vRiU+eCny7lD2vmFZy79Iuw1U=
```
Comment (techcobweb) [inline build-images/github-webhook-receiver/go.sum:31]: This file shouldn't have been changed by this PR, please revert it to what it should have been

**Review by techcobweb — CHANGES_REQUESTED** [review]
Nice new unit tests.
Changes needed before merging.

**Comment by galasa-team** [comment]
Build failed, see http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/galasa-build/pipelineruns/repo-automation-pr-269-hmmzf for details. If you are unable to do so, please contact a member of the Galasa team.

**Comment by galasa-team** [comment]
Build failed, see http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/galasa-build/pipelineruns/repo-automation-pr-269-9hg55 for details. If you are unable to do so, please contact a member of the Galasa team.

File: build-images/github-webhook-receiver/go.sum
Comment (Akyiaa) [inline build-images/github-webhook-receiver/go.sum:1]: reverted to commit 482f39a
`git checkout 482f39a -- build-images/github-webhook-receiver/go.sum  `

File: build-images/github-webhook-receiver/go.sum (hunk 7 above)
Comment (Akyiaa) [reply build-images/github-webhook-receiver/go.sum]: reverted to commit https://github.com/galasa-dev/automation/commit/482f39a54d4a2a66e7f55f963d4b21bc2e4713a3
git checkout 482f39a -- build-images/github-webhook-receiver/go.sum

Comment (Akyiaa) [reply build-images/github-webhook-receiver/go.sum]: reverted to commit https://github.com/galasa-dev/automation/commit/482f39a54d4a2a66e7f55f963d4b21bc2e4713a3
which removed the 706 additions

File: offline-tools/copyrighter/.gitignore (hunk 5 above)
Comment (Akyiaa) [reply offline-tools/copyrighter/.gitignore]: resolved conflicts to accept changes from origin/main

**Comment by galasa-team** [comment]
Build failed, see http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/galasa-build/pipelineruns/repo-automation-pr-269-hgjtt for details. If you are unable to do so, please contact a member of the Galasa team.

**Comment by galasa-team** [comment]
Build failed, see http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/galasa-build/pipelineruns/repo-automation-pr-269-cd5kx for details. If you are unable to do so, please contact a member of the Galasa team.

**Comment by galasa-team** [comment]
Build successful

**Pull request merged**